### DISCUSSION TITLE
Table refresh after broadcast Notification through Pusher

### DISCUSSION THREAD

**Discussion by mschramade** [filamentphp/filament#7679] [Help]
On a resource page I've added an Action which queues a long running process in the background.
When this process ends it will send a broadcast notification through Pusher which I see popup in my app.

At this moment I want to refresh the table data on screen because some values have changed by the background process.

I've tried adding an action to the broadcast notification but I'am not sure what to put in this action to get the table to refresh...

Of course I can poll the table but this is not the way I want to go

Someone any ideas?

**Comment by pboivin** [comment]
There's built-in support for Echo in Livewire, maybe this can help:
https://livewire.laravel.com/docs/events#real-time-events-using-laravel-echo

**Reply by pboivin** [reply]
I think calling any Livewire method on the Page component should refresh the table.

**Reply by mschramade** [reply]
I thought Filament already uses Echo underneath for its broadcast notifications. I don't know how to get to the Page component and call a livewire method on it. All new stuff to me...

**Reply by pboivin** [reply]
You're right, Filament uses Echo for broadcast notifications. I'm not sure if it's possible to execute some arbitrary action along with a Filament notification, that's why I thought of a Livewire method.

The idea would be to add a method on your page component (e.g. ListPosts). Add the `#[On(...)]`attribute to the method to bind it to your Echo event. I think even an empty method would be enough to refresh the table.

**Reply by mschramade** [reply]
Problem with actions on notification windows is that the action is triggered by a button. I want the popped-up broadcast notification to immediately dispatch a Livewire event, not on a button-click.

Fortunately you can use a custom notification view in Filament: 

https://filamentphp.com/docs/3.x/notifications/customizing-notifications#using-a-custom-notification-view

As a template I've used the default view in vendor\filament\notifications\resources\views\notification.blade.php and added:

`<div x-init="$dispatch('myRefreshEvent')"/>`

In the list blade file I've put a listener:

`<div x-on:myRefreshEvent.window="$wire.refreshData()"/>`

this will force Livewire to call the public function refreshData() in the custom php filament component and force a screen update.

This works but seems a bit complicated... I've tried many other ways, also the suggested #[On(...)] attribute, but could not get a dispatched event to this public method with the On attribute, this is the only way I could get it working.

I've also tried calling directly the $wire.refreshData() from the modified Notification blade file, but got php error: Public method [refreshData] not found on component

So for now I've found a work-around, maybe someone has a better way?

**Reply by mariusjp** [reply]
~~Does this still work? I can't seem to get it working on Filament 3.2~~
Figured it out. Two things:
- you can't use camelCase anymore (~~myRefreshEvent~~ => my-refresh-event)
- `$wire.refreshData()` needs to be changed to `$wire.call('$refresh')`

Thanks for figuring this out!

**Comment by hannanmiah** [comment]
Is there any specific solution related to refresh table on a specific broadcasted event?

**Comment by juliancc** [comment]
I got it working and it's pretty easy. Obviously you need to enable broadcast in your notification using `->broadcast()` and then in your filament component you want to listen to the broadcasted Notification  like this:

```php
public function getListeners()
{
        $userId = Auth::id();
        return [
            "echo-private:App.Models.User.{$userId},.Illuminate\\Notifications\\Events\\BroadcastNotificationCreated" => '$refresh',
        ];
}
```
I'm using pusher so I went to their debug console and saw the event being triggered.
